src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -164,606 +136,64 @@ void AsyncRPCOperation_sendmany::main() {
-            "senders and amounts. THIS MAY AFFECT YOUR PRIVACY. Resubmit "
-            "with the `privacyPolicy` parameter set to `AllowRevealedSenders` "
-            "or weaker if you wish to allow this transaction to proceed anyway.");
-    }
-
-    if (recipientPools_.count(OutputPool::Transparent) && !strategy_.AllowRevealedRecipients()) {
-        throw JSONRPCError(
-            RPC_INVALID_PARAMETER,
-            "This transaction would have transparent recipients, which is not "
-            "enabled by default because it will publicly reveal transaction "
-            "recipients and amounts. THIS MAY AFFECT YOUR PRIVACY. Resubmit "
-            "with the `privacyPolicy` parameter set to `AllowRevealedRecipients` "
-            "or weaker if you wish to allow this transaction to proceed anyway.");
-    }
-
-    if (!spendable.sproutNoteEntries.empty()) {
-        if (recipientPools_.count(OutputPool::Sapling) && !strategy_.AllowRevealedAmounts()) {
-            throw JSONRPCError(
-                RPC_INVALID_PARAMETER,
-                "Sending from the Sprout shielded pool to the Sapling "
-                "shielded pool is not enabled by default because it will "
-                "publicly reveal the transaction amount. THIS MAY AFFECT YOUR PRIVACY. "
-                "Resubmit with the `privacyPolicy` parameter set to `AllowRevealedAmounts` "
-                "or weaker if you wish to allow this transaction to proceed anyway.");
-        }
-    }
-
-    if (!spendable.saplingNoteEntries.empty()) {
-        if (recipientPools_.count(OutputPool::Orchard) && !strategy_.AllowRevealedAmounts()) {
-            throw JSONRPCError(
-                RPC_INVALID_PARAMETER,
-                "Sending from the Sapling shielded pool to the Orchard "
-                "shielded pool is not enabled by default because it will "
-                "publicly reveal the transaction amount. THIS MAY AFFECT YOUR PRIVACY. "
-                "Resubmit with the `privacyPolicy` parameter set to `AllowRevealedAmounts` "
-                "or weaker if you wish to allow this transaction to proceed anyway.");
-        }
-        // Sending from Sapling to transparent will be caught above in the
-        // AllowRevealedRecipients check; sending to Sprout is disallowed
-        // entirely.
-    }
-
-    if (!spendable.orchardNoteMetadata.empty()) {
-        if (recipientPools_.count(OutputPool::Sapling) && !strategy_.AllowRevealedAmounts()) {
-            throw JSONRPCError(
-                RPC_INVALID_PARAMETER,
-                "Sending from the Orchard shielded pool to the Sapling "
-                "shielded pool is not enabled by default because it will "
-                "publicly reveal the transaction amount. THIS MAY AFFECT YOUR PRIVACY. "
-                "Resubmit with the `privacyPolicy` parameter set to `AllowRevealedAmounts` "
-                "or weaker if you wish to allow this transaction to proceed anyway.");
-        }
```

Comment:
Right, this is what `InsufficientSaplingFunds` _should_ say (although my changes to that are a bit more general, as the check really only asserts that it can’t do Sapling->Sapling, not that the sending pool is specifically Orchard).